File: app/src/main/java/app/deadmc/devnetworktool/activities/FullViewActivity.java
package app.deadmc.devnetworktool.activities;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.webkit.WebView;
import android.widget.ScrollView;
import android.widget.TextView;
import android.support.v7.widget.Toolbar;

import app.deadmc.devnetworktool.R;
import app.deadmc.devnetworktool.helpers.FileFormatHelper;
import app.deadmc.devnetworktool.helpers.StringHelper;
import app.deadmc.devnetworktool.models.ResponseDev;
import app.deadmc.devnetworktool.singletons.SharedData;

public class FullViewActivity extends AppCompatActivity {

    private ResponseDev responseDev;
    private ScrollView scrollView;
    private TextView textView;
    private WebView webView;
    private Toolbar toolbar;

    private String text;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_full_view);
        initValues();
        initElements();
        setViewDependsOnType();
    }

    public void initValues() {
        responseDev = SharedData.getInstance().getResponseDev();
        Bundle bundle = getIntent().getExtras();
        if (bundle != null && bundle.containsKey("text")) {
            text = bundle.getString("text");
            return;
        }

        if (responseDev != null) {
            text = responseDev.getBody();
        }

    }

    public void initElements() {
        if (text == null)
            return;
        textView = (TextView) findViewById(R.id.jsonTextView);
        scrollView = (ScrollView) findViewById(R.id.scrollView);
        webView = (WebView) findViewById(R.id.webView);
        initToolbar();
        textView.setText(StringHelper.formatString(text));
        setViewDependsOnType();
    }

    private void initToolbar() {
        toolbar = (Toolbar) findViewById(R.id.toobar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setDisplayShowHomeEnabled(true);
    }

    public void setViewDependsOnType() {
        if (text == null) {
            return;
        }

        int type = FileFormatHelper.getTypeOfString(text);
        webView.setVisibility(View.GONE);
        scrollView.setVisibility(View.GONE);
        switch (type) {
            case FileFormatHelper.JSON:
                toolbar.setTitle(R.string.json);
                scrollView.setVisibility(View.VISIBLE);
                textView.setText(StringHelper.formatString(text));
                break;
            case FileFormatHelper.XML:
                toolbar.setTitle(R.string.xml);
                scrollView.setVisibility(View.VISIBLE);
                textView.setText(StringHelper.formatString(text));
                break;
            case FileFormatHelper.HTML:
                toolbar.setTitle(R.string.html);
                webView.setVisibility(View.VISIBLE);
                webView.loadData(text, "text/html; charset=utf-8", "UTF-8");
                //webView.loadUrl("http://google.com");
                break;
            case FileFormatHelper.UNDEFINED:
                scrollView.setVisibility(View.VISIBLE);
                textView.setText(StringHelper.formatString(text));
                break;
        }
    }

    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return true;
    }
}

File: app/src/main/java/app/deadmc/devnetworktool/activities/FullViewActivity.kt
package app.deadmc.devnetworktool.activities

import android.annotation.SuppressLint
import android.os.Bundle
import android.view.View
import android.support.v7.widget.Toolbar
import app.deadmc.devnetworktool.R
import app.deadmc.devnetworktool.constants.FULL_VIEW
import app.deadmc.devnetworktool.helpers.formatString
import app.deadmc.devnetworktool.helpers.safe
import app.deadmc.devnetworktool.interfaces.views.FullView
import app.deadmc.devnetworktool.presenters.FullViewPresenter
import com.arellomobile.mvp.presenter.InjectPresenter
import com.arellomobile.mvp.presenter.PresenterType
import kotlinx.android.synthetic.main.activity_full_view.*

class FullViewActivity : BaseActivity(), FullView {

    private var toolbar: Toolbar? = null

    @InjectPresenter(type = PresenterType.GLOBAL, tag = FULL_VIEW)
    lateinit var fullViewPresenter: FullViewPresenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_full_view)
        initElements()
    }

    fun initElements() {
        initToolbar()
        fullViewPresenter.setViewByType()
    }

    private fun initToolbar() {
        safe {
            toolbar = findViewById<View>(R.id.toobar) as Toolbar
            setSupportActionBar(toolbar)
            supportActionBar!!.setDisplayHomeAsUpEnabled(true)
            supportActionBar!!.setDisplayShowHomeEnabled(true)
        }
    }

    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return true
    }

    override fun hide() {
        webView.visibility = View.GONE
        scrollView.visibility = View.GONE
    }

    override fun setResult(stringId: Int, text: String) {
        toolbar?.setTitle(stringId)
        setTitle(stringId)
        scrollView.visibility = View.VISIBLE
        jsonTextView.text = formatString(text)
    }

    @SuppressLint("SetJavaScriptEnabled")
    override fun setResultWebView(stringId: Int, text: String, url:String) {
        toolbar?.setTitle(stringId)
        setTitle(stringId)
        webView.settings.javaScriptEnabled = true
        webView.visibility = View.VISIBLE
        val mimeType = "text/html"
        val encoding = "UTF-8"
        webView.loadDataWithBaseURL(url, text, mimeType, encoding, url)
    }
}
